src/corecel/cont/Bitset.hh

```diff
@@ -0,0 +1,395 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/cont/Bitset.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <climits>
+
+#include "corecel/Config.hh"
+
+#include "corecel/Assert.hh"
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/cont/detail/BitsetUtils.hh"
+#include "corecel/math/Algorithms.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Device-compatible bitset implementation.
+ *
+ * This implementation is based on libstdc++'s std::bitset implementation.
+ * It it a subset of the C++ standard, but it should be sufficient
+ * for our current use case. Given that GPU typically use 32-bit words, this
+ * uses unsigned int as the word type instead of the unsigned long used by the
+ * standard library. This container is not thread-safe, multiple threads are
+ * likely to manipulate the same word and access is not synchronized.
+ *
```

Comment:
Since all the methods are defined inline the class definition, the class is hard to read. Can you either un-inline those classes or provide a synopsis in the class documentation here?